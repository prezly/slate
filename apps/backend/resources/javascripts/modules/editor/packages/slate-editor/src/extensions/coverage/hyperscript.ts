/* eslint-disable @typescript-eslint/no-namespace */

import type { CoverageEntry } from '@prezly/sdk';
import { COVERAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-coverage': {
                children?: ReactNode;
                coverage: {
                    id: CoverageEntry['id'];
                };
                uuid: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-coverage': { type: COVERAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
