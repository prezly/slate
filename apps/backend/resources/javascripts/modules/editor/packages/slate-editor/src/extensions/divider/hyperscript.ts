/* eslint-disable @typescript-eslint/no-namespace */

import { DIVIDER_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
