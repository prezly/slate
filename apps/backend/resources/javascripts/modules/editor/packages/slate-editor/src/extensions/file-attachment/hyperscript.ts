/* eslint-disable @typescript-eslint/no-namespace */

import { ATTACHMENT_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-file-attachment': {
                children?: ReactNode;
                description: string;
                file: UploadedFile;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-file-attachment': { type: ATTACHMENT_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
