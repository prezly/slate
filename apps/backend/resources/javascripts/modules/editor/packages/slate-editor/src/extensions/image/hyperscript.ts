/* eslint-disable @typescript-eslint/no-namespace */

import type { ImageLayout } from '@prezly/slate-types';
import { IMAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadedFile } from '@prezly/uploadcare';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-image': {
                children?: ReactNode;
                file: UploadedFile;
                href: string;
                layout: ImageLayout;
                width: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-image': { type: IMAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
