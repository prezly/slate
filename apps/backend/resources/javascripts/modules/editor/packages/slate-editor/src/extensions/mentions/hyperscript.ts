/* eslint-disable @typescript-eslint/no-namespace */

import type { VariableNode } from '@prezly/slate-types';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { VARIABLE_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-placeholder-mention': {
                children?: ReactNode;
                key: VariableNode['key'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-placeholder-mention': { type: VARIABLE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
