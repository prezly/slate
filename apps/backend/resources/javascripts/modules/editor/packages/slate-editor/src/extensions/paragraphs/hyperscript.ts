/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
    },
});
