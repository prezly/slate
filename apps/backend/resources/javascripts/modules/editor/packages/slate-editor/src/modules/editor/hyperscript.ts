/* eslint-disable @typescript-eslint/no-namespace */

import {
    BULLETED_LIST_NODE_TYPE,
    DIVIDER_NODE_TYPE,
    LIST_ITEM_NODE_TYPE,
    LIST_ITEM_TEXT_NODE_TYPE,
    NUMBERED_LIST_NODE_TYPE,
    PARAGRAPH_NODE_TYPE,
    QUOTE_NODE_TYPE,
} from '@prezly/slate-types';
import type { ReactNode } from 'react';
import { createHyperscript, createText } from 'slate-hyperscript';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-li': {
                children?: ReactNode;
            };
            'h-li-text': {
                children?: ReactNode;
            };
            'h-ol': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
            'h-ul': {
                children?: ReactNode;
            };
            'h:blockquote': {
                children?: ReactNode;
            };
            'h:text': {
                children?: ReactNode;
            };
        }
    }
}

export const hyperscript = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
        'h-divider': { type: DIVIDER_NODE_TYPE },
        'h-ol': { type: NUMBERED_LIST_NODE_TYPE },
        'h-ul': { type: BULLETED_LIST_NODE_TYPE },
        'h-li': { type: LIST_ITEM_NODE_TYPE },
        'h-li-text': { type: LIST_ITEM_TEXT_NODE_TYPE },
        'h:blockquote': { type: QUOTE_NODE_TYPE },
    },
    creators: {
        'h-text': createText,
        'h:text': createText,
    },
});
