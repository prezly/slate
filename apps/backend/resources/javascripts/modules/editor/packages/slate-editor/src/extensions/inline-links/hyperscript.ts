/* eslint-disable @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars */
import { LINK_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { createEditor as createSlateEditor } from 'slate';
import { createHyperscript, createEditor as createEditorFactory } from 'slate-hyperscript';

import { createEditor } from '#modules/editor';

import { InlineLinksExtension } from './InlineLinksExtension';

// This allows tests to include Slate Nodes written in JSX without TypeScript complaining.
declare global {
    namespace JSX {
        interface IntrinsicElements {
            [elemName: string]: any; // eslint-disable-line
        }
    }
}

const extensions = [InlineLinksExtension()];

export const hyperscript = createHyperscript({
    elements: {
        link: { type: LINK_NODE_TYPE },
        paragraph: { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        editor: createEditorFactory(() => createEditor(createSlateEditor(), () => extensions)),
    },
});
