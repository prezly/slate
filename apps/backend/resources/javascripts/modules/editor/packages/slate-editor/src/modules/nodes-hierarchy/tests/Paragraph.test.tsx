/** @jsx hyperscript */

import { Alignment } from '@prezly/slate-types';
import { Editor } from 'slate';

import { hyperscript } from '../../../hyperscript';

describe('nodes-hierarchy / Paragraph', () => {
    it('should be kept after normalization', function () {
        const editor = (
            <editor>
                <h:paragraph align={Alignment.LEFT}>
                    <h:text>Left</h:text>
                </h:paragraph>
                <h:paragraph align={Alignment.CENTER}>
                    <h:text>Center</h:text>
                </h:paragraph>
                <h:paragraph align={Alignment.RIGHT}>
                    <h:text>Right</h:text>
                </h:paragraph>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:paragraph align={Alignment.LEFT}>
                    <h:text>Left</h:text>
                </h:paragraph>
                <h:paragraph align={Alignment.CENTER}>
                    <h:text>Center</h:text>
                </h:paragraph>
                <h:paragraph align={Alignment.RIGHT}>
                    <h:text>Right</h:text>
                </h:paragraph>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });

    it('should properly normalize deeply nested paragraphs', () => {
        const editor = (
            <editor>
                <paragraph>
                    <h:text>Bonjour Sylvain, </h:text>
                </paragraph>
                <paragraph>
                    <paragraph>
                        <h:text></h:text>
                    </paragraph>
                </paragraph>
                <paragraph>
                    <h:text>{"J'espère que vous allez bien ?"}</h:text>
                    <paragraph>
                        <h:text></h:text>
                    </paragraph>
                    <paragraph>
                        <paragraph>
                            <h:text></h:text>
                        </paragraph>
                    </paragraph>
                    <paragraph>
                        <h:text>
                            Je me permets de vous contacter car je souhaitais vous proposer de vous
                            entretenir avec{' '}
                        </h:text>
                        <link href="https://www.linkedin.com/in/bernhardney/">
                            <h:text bold>Bernhard Ney</h:text>
                        </link>
                        <h:text bold>, Directeur International de </h:text>
                        <link href="https://www.comatch.com/fr/">
                            <h:text bold>COMATCH</h:text>
                        </link>
                        <h:text bold>
                            , premier réseau de consultants et experts indépendants en Europe.{' '}
                        </h:text>
                    </paragraph>
                    <paragraph>
                        <paragraph>
                            <paragraph>
                                <h:text></h:text>
                            </paragraph>
                        </paragraph>
                        <paragraph>
                            <h:text>
                                {
                                    "A la veille du reconfinement, je vous propose de faire le point avec lui sur l'industrie du conseil et le métier de consultant, qui passe traditionnellement beaucoup de temps en déplacement pour rencontrer ses clients et réaliser ses missions. Voici quelques sujets que vous pourriez aborder avec Bernhard:"
                                }
                            </h:text>
                        </paragraph>
                        <paragraph>
                            <paragraph>
                                <h:text>
                                    {
                                        "A cette occasion, il pourra également dévoiler les résultats de l'étude "
                                    }
                                </h:text>
                                <h:text italic>
                                    {
                                        'L’ADN du consultant indépendant en 2020 : Résilience et flexibilité en temps de crise'
                                    }
                                </h:text>
                                <h:text>
                                    {
                                        " (enquête réalisé avant et après le début de la crise), sur les limites et tendances du conseil indépendant en cette période d'incertitudes économiques. "
                                    }
                                </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                            </paragraph>
                            <paragraph>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                            </paragraph>
                            <paragraph>
                                <h:text bold>
                                    {
                                        "Ce pourrait également être l'occasion de faire le point sur COMATCH, son modèle et sa présence en France. "
                                    }
                                </h:text>
                            </paragraph>
                            <paragraph>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                            </paragraph>
                            <paragraph>
                                <h:text bold>COMATCH en bref</h:text>
                            </paragraph>
                            <paragraph>
                                <h:text>
                                    {
                                        "Le rôle de COMATCH est de mettre les entreprises - de la PME au grand groupes - en relation avec les profils adaptés à leurs besoins. Chacun des 10,000 profils internationaux inscrits sur le réseau a suivi un processus de sélection rigoureux pour assurer le meilleur service aux clients. Aujourd'hui le réseau enregistre plus de 10000 consultants principalement d'Europe et des Etats-Unis, mais également d'Asie ou encore d'Amérique du Sud. "
                                    }
                                </h:text>
                            </paragraph>
                            <paragraph>
                                <h:text>
                                    COMATCH répond ainsi à la quête de réactivité et de flexibilité
                                    des entreprises dans leur recherche en compétences externes et
                                    notamment dans le contexte actuel.{' '}
                                </h:text>
                            </paragraph>
                            <paragraph>
                                <h:text> </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text>Est-ce que cette proposition vous intéresse ?</h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text> </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text>
                                    {
                                        "Dans l'attente de votre retour, je vous souhaite une excellente fin de journée."
                                    }
                                </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text> </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text>Bien cordialement,</h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text> </h:text>
                                <paragraph>
                                    <h:text></h:text>
                                </paragraph>
                                <h:text>Noémie Diaz de Cerio</h:text>
                            </paragraph>
                        </paragraph>
                    </paragraph>
                </paragraph>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <paragraph>
                    <h:text>Bonjour Sylvain, </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>{"J'espère que vous allez bien ?"}</h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        Je me permets de vous contacter car je souhaitais vous proposer de vous
                        entretenir avec{' '}
                    </h:text>
                    <link href="https://www.linkedin.com/in/bernhardney/">
                        <h:text bold>Bernhard Ney</h:text>
                    </link>
                    <h:text bold>, Directeur International de </h:text>
                    <link href="https://www.comatch.com/fr/">
                        <h:text bold>COMATCH</h:text>
                    </link>
                    <h:text bold>
                        , premier réseau de consultants et experts indépendants en Europe.{' '}
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        {
                            "A la veille du reconfinement, je vous propose de faire le point avec lui sur l'industrie du conseil et le métier de consultant, qui passe traditionnellement beaucoup de temps en déplacement pour rencontrer ses clients et réaliser ses missions. Voici quelques sujets que vous pourriez aborder avec Bernhard:"
                        }
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        {"A cette occasion, il pourra également dévoiler les résultats de l'étude "}
                    </h:text>
                    <h:text italic>
                        {
                            'L’ADN du consultant indépendant en 2020 : Résilience et flexibilité en temps de crise'
                        }
                    </h:text>
                    <h:text>
                        {
                            " (enquête réalisé avant et après le début de la crise), sur les limites et tendances du conseil indépendant en cette période d'incertitudes économiques. "
                        }
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text bold>
                        {
                            "Ce pourrait également être l'occasion de faire le point sur COMATCH, son modèle et sa présence en France. "
                        }
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text bold>COMATCH en bref</h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        {
                            "Le rôle de COMATCH est de mettre les entreprises - de la PME au grand groupes - en relation avec les profils adaptés à leurs besoins. Chacun des 10,000 profils internationaux inscrits sur le réseau a suivi un processus de sélection rigoureux pour assurer le meilleur service aux clients. Aujourd'hui le réseau enregistre plus de 10000 consultants principalement d'Europe et des Etats-Unis, mais également d'Asie ou encore d'Amérique du Sud. "
                        }
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        COMATCH répond ainsi à la quête de réactivité et de flexibilité des
                        entreprises dans leur recherche en compétences externes et notamment dans le
                        contexte actuel.{' '}
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text> </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>Est-ce que cette proposition vous intéresse ?</h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text> </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>
                        {
                            "Dans l'attente de votre retour, je vous souhaite une excellente fin de journée."
                        }
                    </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text> </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>Bien cordialement,</h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text> </h:text>
                </paragraph>
                <paragraph>
                    <h:text></h:text>
                </paragraph>
                <paragraph>
                    <h:text>Noémie Diaz de Cerio</h:text>
                </paragraph>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });
});
