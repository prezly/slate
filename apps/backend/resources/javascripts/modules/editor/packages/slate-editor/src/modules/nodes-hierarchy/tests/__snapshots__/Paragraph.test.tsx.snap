// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodes-hierarchy / Paragraph should be kept after normalization 1`] = `
[
  {
    "align": "left",
    "children": [
      {
        "text": "Left",
      },
    ],
    "type": "paragraph",
  },
  {
    "align": "center",
    "children": [
      {
        "text": "Center",
      },
    ],
    "type": "paragraph",
  },
  {
    "align": "right",
    "children": [
      {
        "text": "Right",
      },
    ],
    "type": "paragraph",
  },
]
`;
